test/app/features/main_menu/domain/repositories/user_profile_repository_test.dart

```diff
@@ -0,0 +1,348 @@
+import 'package:dartz/dartz.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:penhas/app/core/entities/valid_fiel.dart';
+import 'package:penhas/app/core/error/failures.dart';
+import 'package:penhas/app/core/network/api_client.dart';
+import 'package:penhas/app/core/network/api_server_configure.dart';
+import 'package:penhas/app/features/main_menu/data/model/account_preference_model.dart';
+import 'package:penhas/app/features/main_menu/domain/repositories/user_profile_repository.dart';
+
+class MockApiProvider extends Mock implements IApiProvider {}
+
+class MockApiServerConfigure extends Mock implements IApiServerConfigure {}
+
+void main() {
+  late IUserProfileRepository sut;
+  late IApiProvider apiProvider;
+  late IApiServerConfigure serverConfiguration;
+
+  setUpAll(() {
+    apiProvider = MockApiProvider();
+    serverConfiguration = MockApiServerConfigure();
+    sut = UserProfileRepository(
+      apiProvider: apiProvider,
+      serverConfiguration: serverConfiguration,
+    );
+  });
+
+  void _setUpMockPost() {
+    when(
+      () => apiProvider.post(
+        path: any(named: 'path'),
+        parameters: any(named: 'parameters'),
+      ),
+    ).thenAnswer((_) async => Future.value(''));
+  }
+
+  void _setUpMockPostException() {
+    when(
+      () => apiProvider.post(
+        path: any(named: 'path'),
+        parameters: any(named: 'parameters'),
+      ),
+    ).thenThrow(Exception());
+  }
+
+  group(UserProfileRepository, () {
+    group('stealth mode', () {
+      test(
+        'hit endpoint /me/modo-camuflado-toggle',
+        () async {
+          // arrange
+          _setUpMockPost();
+          final endPoint = ['me', 'modo-camuflado-toggle'].join('/');
```

Comment:
Em alguns testes o endpoint está diretamente com o path e em outros está assim, fazendo o join de uma lista, acha que poderia deixar todos diretamente com o path?

```suggestion
          final endPoint = '/me/modo-camuflado-toggle';
```